src/web/common.rkt

```diff
@@ -77,118 +78,166 @@
+  (if (supported-by-lang? prog* "tex")
+      (core->tex prog* #:loc (and loc (cons 2 loc)) #:color "blue")
+      "ERROR"))
 
-(define (preprocess->java preprocess)
-  (match preprocess
-    [(list 'sort variables ...)
-     (format "assert ~a;" (format-less-than-condition variables))]
+(define (combine-fpcore-instruction i e c)
+  (match i
     [(list 'abs x)
-     (format "~a = Math.abs(~a);" x x)]))
-
-(define (preprocess->python preprocess)
-  (match preprocess
-    [(list 'sort variables ...)
-     (define comma-joined (string-join (map ~a variables) ", "))
-     (format "[~a] = sort([~a])" comma-joined comma-joined)]
+     (define x* (string->symbol (string-append (symbol->string x) "_m")))
+     (define e* (replace-vars (list (cons x x*)) e))
+     (define p (index-of (context-vars c) x))
+     (define c* (struct-copy context c [vars (list-set (context-vars c) p x*)]))
+     (cons e* c*)]
+    [(list 'negabs x)
+     (define x-string (symbol->string x))
+     (define x-sign (string->symbol (string-append x-string "_s")))
+     (define x* (string->symbol (string-append x-string "_m")))
+     (define p (index-of (context-vars c) x))
+     (define r (list-ref (context-var-reprs c) p))
+     (define c* (struct-copy context c [vars (list-set (context-vars c) p x*)]))
+     (define c** (context-extend c* x-sign r))
+     (define e*
+       (list
+        (get-parametric-operator '* r (context-repr c))
+        x-sign
+        (replace-vars (list (cons x x*)) e)))
+     (cons e* c**)]
+    [_
+     (cons e c)]))
+
+(define (format-prelude-instruction instruction ctx ctx* language converter)
+  (define (converter* e c)
+    (define fpcore (program->fpcore e c))
+    (define output (converter fpcore "code"))
+    (define lines (string-split output "\n"))
+    (match language
+      ["FPCore" (pretty-format e #:mode 'display)]
+      ["Fortran" (string-trim (third lines) #px"\\s+code\\s+=\\s+")]
+      ["MATLAB" (string-trim (second lines) #px"\\s+tmp\\s+=\\s+")]
+      ["Wolfram" (string-trim (first lines) #px".*:=\\s+")]
+      ["TeX" output]
+      [_ (string-trim (second lines) #px"\\s+return\\s+")]))
+  (match instruction
     [(list 'abs x)
-     (format "~a = abs(~a)" x x)]))
-
-(define (preprocess->julia preprocess)
-  (match preprocess
-    [(list 'sort variables ...)
-      (define comma-joined (string-join (map ~a variables) ", "))
+     (define x* (string->symbol (string-append (symbol->string x) "_m")))
+     (define r (list-ref (context-var-reprs ctx) (index-of (context-vars ctx) x)))
+     (define e (list (get-parametric-operator 'fabs r) x))
+     (define c (context (list x*) r r))
+     (format "~a = ~a" x* (converter* e c))]
+    [(list 'negabs x)
+     (define x-string (symbol->string x))
+     (define x* (string->symbol (string-append x-string "_m")))
+     (define r (list-ref (context-var-reprs ctx) (index-of (context-vars ctx) x)))
+     (define e* (list (get-parametric-operator 'fabs r) x))
```

Comment:
You probably could have saved yourself some code here by:
- Moving the string munging code to its own `(var-extend x 'm)` helper
- Generating FPCore instead of Herbie internals, that way you can skip the `get-parametric-operator`
But I think it's fine, we can clean up later / separate PR. It was ugly before, too!